test/integration/locale_test.rb

```diff
@@ -0,0 +1,22 @@
+require "test_helper"
+
+class LocaleTest < SystemTest
+  test "html lang attribute is set from locale" do
+    I18n.available_locales.each do |locale|
+      visit root_path(locale: locale)
+      assert_equal locale.to_s, page.find("html")[:lang]
+    end
+  end
+
+  test "locale is switched via locale menu" do
+    visit root_path
+    assert_equal I18n.default_locale.to_s, page.find("html")[:lang]
+
+    click_link "Deutsch"
+    assert_equal "de", page.find("html")[:lang]
+  end
+
+  teardown do
+    visit root_path(locale: I18n.default_locale)
```

Comment:
I am not sure I understand what value this teardown is adding. The above two tests would work the same without this.

Reply:
Without the teardown, It seems the locale selection is persisted between `SystemTest` runs.

So in the tests I've provided, the first test leaves the locale on `:ja`, which cases the second test to fail, and either test on their own would cause all subsequent system tests to fail as the buttons and form labels no longer contain the English test expected by the other tests.

Example:

```
$ rails test test/integration/locale_test.rb test/integration/profile_test.rb --seed 60333

# Running:

.F

Failure:
LocaleTest#test_locale_is_switched_via_locale_menu [/Users/jsrn/Code/personal/rubygems.org/test/integration/locale_test.rb:13]:
--- expected
+++ actual
@@ -1,3 +1 @@
-# encoding: US-ASCII
-#    valid: true
-"en"
+"ja"

rails test test/integration/locale_test.rb:11

E

Error:
ProfileTest#test_changing_to_an_existing_handle:
Capybara::ElementNotFound: Unable to find visible field "Email or Username" that is not disabled
    test/integration/profile_test.rb:10:in `sign_in'
    test/integration/profile_test.rb:32:in `block in <class:ProfileTest>'
```

Is there a better way to handle this?

Reply:
got it. thanks for explaining. please use `teardown { reset_session! }`. It makes it clear that we want to reset cookies in the teardown.

Reply:
Ah, brilliant, thanks! :)

I've updated the PR.